pkg/upgrade/upgrade.go

```diff
@@ -287,6 +290,9 @@ func CleanupExistingResource(ctx context.Context,
 	toDelete := getDashboardWatsonResources(dscApplicationsNamespace)
 	multiErr = multierror.Append(multiErr, deleteResources(ctx, cli, &toDelete))
 
+	// cleanup nvidia nim integration remove tech preview
+	multiErr = multierror.Append(multiErr, cleanupNimIntegrationTechPreview(ctx, cli, oldReleaseVersion))
```

Comment:
@andrewballantyne dont we need to cleanup api key secret also? AFAIU UI will create the secret when user selects the enable button. So, if the secret is present then it will result in an error.

Reply:
Yeah, that's a good catch. If we don't delete the secret and we don't have a NIM Account CR here... it will likely block the creation flow. @etirelli should we delete the customer's "key file" or look at some what to keep/update it on install in the new Controller way?

Reply:
I would say update/reuse would be best as it is unlikely the API key itself changed and forcing the customer to update might create confusion.

Reply:
discussed in slack. We will delete the secret if it exists and let the user recreate it.

Reply:
Resolved in https://github.com/opendatahub-io/opendatahub-operator/pull/1369/commits/0483894b856ac71e7ff3154ecc37f35f46ae1364.